pkg/clustermanager/cluster_manager.go

```diff
@@ -708,16 +708,23 @@ func (c *ClusterManager) waitForControlPlaneReplicasReady(ctx context.Context, m
 }
 
 func (c *ClusterManager) waitForMachineDeploymentReplicasReady(ctx context.Context, managementCluster *types.Cluster, clusterSpec *cluster.Spec) error {
+	var err error
```

Comment:
You don't really need to declare the error up here - its rather abnormal for Go.